scripts/irods/test/test_targeting_specific_replica_number.py

```diff
@@ -0,0 +1,356 @@
+
+        super(test_targeting_specific_replica_number__issue_6896, self).tearDown()
+
+
+    def test_iget_n_returns_error_when_requested_replica_votes_0(self):
+        local_iget_file = self.local_file + '_from_iget'
+
+        if os.path.exists(local_iget_file):
+            os.unlink(local_iget_file)
+
+        # Overwrite the first replica with a different file so that the replicas can be differentiated.
+        self.user.assert_icommand(['iput', '-f', '-n0', self.other_local_file, self.logical_path])
+        self.assertEqual(
+            str(self.other_file_size), lib.get_replica_size(self.user, os.path.basename(self.logical_path), 0))
+        self.assertEqual(str(self.file_size), lib.get_replica_size(self.user, os.path.basename(self.logical_path), 1))
+        self.assertEqual(str(1), lib.get_replica_status(self.user, os.path.basename(self.logical_path), 0))
+        self.assertEqual(str(0), lib.get_replica_status(self.user, os.path.basename(self.logical_path), 1))
+
+        # set read weight to 0.0 on passthru with replica 1
+        self.admin.assert_icommand(['iadmin', 'modresc', self.pt2, 'context', 'read=0.0'])
+
+        self.user.assert_icommand(['ils', '-L', self.logical_path], 'STDOUT') # debugging
+
+        # execute iget -n 1, assert that an error is returned because the requested replica voted 0.
+        self.user.assert_icommand(
+            ['iget', '-n1', self.logical_path, local_iget_file], 'STDERR', 'SYS_REPLICA_INACCESSIBLE')
+        self.assertFalse(os.path.exists(local_iget_file))
+
+
+    def test_iget_n_always_returns_requested_replica(self):
+        local_iget_file = self.local_file + '_from_iget'
+
+        if os.path.exists(local_iget_file):
+            os.unlink(local_iget_file)
+
+        # Overwrite the first replica with a different file so that the replicas can be differentiated.
+        self.user.assert_icommand(['iput', '-f', '-n0', self.other_local_file, self.logical_path])
+        self.assertEqual(
+            str(self.other_file_size), lib.get_replica_size(self.user, os.path.basename(self.logical_path), 0))
+        self.assertEqual(str(self.file_size), lib.get_replica_size(self.user, os.path.basename(self.logical_path), 1))
+        self.assertEqual(str(1), lib.get_replica_status(self.user, os.path.basename(self.logical_path), 0))
+        self.assertEqual(str(0), lib.get_replica_status(self.user, os.path.basename(self.logical_path), 1))
+
+        self.user.assert_icommand(['ils', '-L', self.logical_path], 'STDOUT') # debugging
+
+        # set read weight to 0.1 (very low) on passthru with replica 1
+        self.admin.assert_icommand(['iadmin', 'modresc', self.pt2, 'context', 'read=0.1'])
+
+        # execute iget -n 1, assert that replica 1 is returned even though it will vote extremely low.
+        self.user.assert_icommand(['iget', '-n1', self.logical_path, local_iget_file])
+        assert_command(['diff', self.local_file, local_iget_file])
```

Comment:
What is `assert_command`? Is that part of the unittest framework or do we provide that?

Reply:
We provide it: https://github.com/irods/irods/blob/ee3a689971435d81e12aef6500d27ff33a9c2a01/scripts/irods/test/command.py#L76-L77

Reply:
I chose `assert_command` because `diff` is not an iCommand, so `self.<some_user>.assert_icommand` seemed unfitting (although I have no reason to believe it wouldn't do the same thing). There is no member of `IrodsSession` which serves an equivalent role, as far as I can tell.